servers/rendering/renderer_viewport.cpp

```diff
@@ -971,8 +971,10 @@ void RendererViewport::viewport_set_use_xr(RID p_viewport, bool p_use_xr) {
 void RendererViewport::viewport_set_scaling_3d_mode(RID p_viewport, RS::ViewportScaling3DMode p_mode) {
 	Viewport *viewport = viewport_owner.get_or_null(p_viewport);
 	ERR_FAIL_NULL(viewport);
-	ERR_FAIL_COND_EDMSG(p_mode == RS::VIEWPORT_SCALING_3D_MODE_FSR && OS::get_singleton()->get_current_rendering_method() != "forward_plus", "FSR1 is only available when using the Forward+ renderer.");
-	ERR_FAIL_COND_EDMSG(p_mode == RS::VIEWPORT_SCALING_3D_MODE_FSR2 && OS::get_singleton()->get_current_rendering_method() != "forward_plus", "FSR2 is only available when using the Forward+ renderer.");
+	const String rendering_method = OS::get_singleton()->get_current_rendering_method();
+	ERR_FAIL_COND_EDMSG(p_mode == RS::VIEWPORT_SCALING_3D_MODE_FSR && rendering_method != "forward_plus", "FSR1 is only available when using the Forward+ renderer.");
+	ERR_FAIL_COND_EDMSG(p_mode == RS::VIEWPORT_SCALING_3D_MODE_FSR2 && rendering_method != "forward_plus", "FSR2 is only available when using the Forward+ renderer.");
+	ERR_FAIL_COND_EDMSG(p_mode == RS::VIEWPORT_SCALING_3D_MODE_METALFX_TEMPORAL && rendering_method != "forward_plus", "MetalFX Temporal is only available when using the Forward+ renderer.");
```

Comment:
Is MetalFX Spatial implemented in the Mobile renderer? I'm curious about it, since we don't have working FSR1 in Mobile right now. (We could probably add it, but FSR1 is rather demanding on mobile devices when it's not in tweaked form like https://github.com/godotengine/godot-proposals/issues/3786 or https://github.com/godotengine/godot-proposals/issues/10411.)

Reply:
Apple's MetalFX are also optimised for their mobile devices, so they would still be usable there, but if the motion vectors are not viable for MetalFX Temporal upscaling, then we won't use it 👍🏻